launcher/minecraft/mod/ShaderPackFolderModel.h

```diff
@@ -9,4 +12,14 @@ class ShaderPackFolderModel : public ResourceFolderModel {
     explicit ShaderPackFolderModel(const QString& dir, BaseInstance* instance) : ResourceFolderModel(QDir(dir), instance) {}
 
     virtual QString id() const override { return "shaderpacks"; }
+
+    [[nodiscard]] Task* createUpdateTask() override
+    {
+        return new BasicFolderLoadTask(m_dir, [](QFileInfo const& entry) { return makeShared<ShaderPack>(entry); });
+    }
+
+    [[nodiscard]] Task* createParseTask(Resource& resource) override
+    {
+        return new LocalShaderPackParseTask(m_next_resolution_ticket, static_cast<ShaderPack&>(resource));
+    }
```

Comment:
How are these involved here? Why wasn't this needed before? Just wondering, this shouldn't block merging

Reply:
it allows using a subclass of Resource and therefore overriding `applyFilter`

Reply:
TheKobeToad explanation is accurate. 
As for why was not used before, I'm not sure as it only looks like the parsing code was used for mods.
There are similar unused tasks for other resources.
The only place it is used is when calling a function to determine the type of a resource( used in curseforge modpack import)